zcnbridge/config.go

```diff
@@ -1,260 +1,97 @@
-	startTime common.Timestamp
+type BridgeSDKConfig struct {
+	LogLevel        *string
+	LogPath         *string
+	ConfigChainFile *string
+	ConfigDir       *string
+	Development     *bool
 }
 
 type BridgeClient struct {
-	*BridgeConfig
-	*BridgeClientConfig
-	*Instance
-}
+	BridgeAddress,
+	TokenAddress,
+	AuthorizersAddress,
+	EthereumAddress,
+	Password,
+	EthereumNodeURL,
+	Homedir string
 
-type BridgeOwner struct {
-	*BridgeClientConfig
-	*Instance
-}
-
-// ReadClientConfigFromCmd reads config from command line
-// Bridge has several configs:
-// Chain config at ~/.zcn/config.json
-// User 0Chain wallet config at ~/.zcn/wallet.json
-// User EthBridge config ~/.zcn/bridge.json
-// Owner EthBridge config ~/.zcn/bridgeowner.json
-func ReadClientConfigFromCmd() *BridgeSDKConfig {
-	// reading from bridge.yaml
-	cmd := &BridgeSDKConfig{}
-	cmd.Development = flag.Bool("development", false, "development mode")
-	cmd.LogPath = flag.String("logs", "./logs", "log folder")
-	cmd.ConfigDir = flag.String("path", GetConfigDir(), "config home folder")
-	cmd.ConfigBridgeFile = flag.String("bridge_config", BridgeClientConfigName, "bridge config file")
-	cmd.ConfigChainFile = flag.String("chain_config", ZChainsClientConfigName, "chain config file")
-	cmd.LogLevel = flag.String("loglevel", "debug", "log level")
-
-	flag.Parse()
-
-	return cmd
-}
-
-func CreateBridgeOwner(cfg *viper.Viper, walletFile ...string) *BridgeOwner {
-	owner := cfg.Get(OwnerConfigKeyName)
-	if owner == nil {
-		ExitWithError("CreateBridgeOwner: can't read config with `owner` key")
-	}
-
-	fileUsed := cfg.ConfigFileUsed()
-	homedir := path.Dir(fileUsed)
-	if homedir == "" {
-		ExitWithError("CreateBridgeOwner: homedir is required")
-	}
-
-	wallet, err := loadWallet(homedir, walletFile...)
-	if err != nil {
-		ExitWithError("Error reading the wallet", err)
-	}
-
-	return &BridgeOwner{
-		BridgeClientConfig: &BridgeClientConfig{
-			ContractsRegistry: ContractsRegistry{
-				BridgeAddress:      cfg.GetString(fmt.Sprintf("%s.BridgeAddress", OwnerConfigKeyName)),
-				WzcnAddress:        cfg.GetString(fmt.Sprintf("%s.WzcnAddress", OwnerConfigKeyName)),
-				AuthorizersAddress: cfg.GetString(fmt.Sprintf("%s.AuthorizersAddress", OwnerConfigKeyName)),
-			},
-			EthereumConfig: EthereumConfig{
-				EthereumNodeURL: cfg.GetString(fmt.Sprintf("%s.EthereumNodeURL", OwnerConfigKeyName)),
-				GasLimit:        cfg.GetUint64(fmt.Sprintf("%s.GasLimit", OwnerConfigKeyName)),
-				Value:           cfg.GetInt64(fmt.Sprintf("%s.Value", OwnerConfigKeyName)),
-			},
-			EthereumAddress: cfg.GetString(fmt.Sprintf("%s.EthereumAddress", OwnerConfigKeyName)),
-			Password:        cfg.GetString(fmt.Sprintf("%s.Password", OwnerConfigKeyName)),
-			Homedir:         homedir,
-		},
-		Instance: &Instance{
-			startTime: common.Now(),
-			zcnWallet: wallet,
-		},
-	}
+	ConsensusThreshold float64
+	GasLimit           uint64
 }
 
 func CreateBridgeClient(cfg *viper.Viper, walletFile ...string) *BridgeClient {
```

Comment:
The walletFile option param is not being used

Reply:
Done